subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -61,32 +61,30 @@ static void xor_128(const uint8_t a[16], const uint8_t b[16], uint8_t out[16])
 	}
 }
 
-int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint32_t r,
-		uint32_t *out)
+int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint8_t r[BT_CSIP_SIH_PRAND_SIZE],
+		uint8_t out[BT_CSIP_SIH_HASH_SIZE])
 {
-	uint8_t res[16]; /* need to store 128 bit */
+	uint8_t res[BT_CSIP_PRAND_SIZE]; /* need to store 128 bit */
```

Comment:
`res` needs to be 16 octets as it is passed to `bt_encrypt_be` which requires 16-octets of plaintext and output buffer

Reply:
Yes, this is an error :)

Reply:
It is.
If you look in the top of the file, I created a new macro (to avoid magic numbering here).
`#define BT_CSIP_PRAND_SIZE          (BT_CSIP_CRYPTO_PADDING_SIZE + BT_CSIP_SIH_PRAND_SIZE)`
I understand the confusion, as you are referring to the already existing one called `BT_CSIP_PRAND_SIZE`

The naming of prand in spec. I think is a little bid ambiguous, it does refer to a 24-bit random number, but it also in samples refers to the 16-octet zero-padded 24-bit random number.
Maybe I should rename `BT_CSIP_PRAND_SIZE` into `BT_CSIP_PADDED_RAND_SIZE`, what do you think?